### PR TITLE
Forward reasoning

### PR DESCRIPTION
**Pull request by ChenfengWei0**
(1) fix multi-property: remove the claims proven correct in forward condition or inductive step. 
(2) fix coverage: incorrect coverage message output format in incr and kind. Also fix the signature for the clear verified algorithm.
(3) refactoring: multi-property-relevant code got refactored.
(4) complete the verbose output

### CODE REVIEW

File: src/goto-programs/goto_coverage.cpp [hunk 1]
```diff
@@ -1,7 +1,10 @@
 #include <goto-programs/goto_coverage.h>
 
-size_t goto_coveraget::total_branch = 0;
+size_t goto_coveraget::total_assert = 0;
```
Comment (lucasccordeiro) [inline src/goto-programs/goto_coverage.cpp:3]: Do we really need these global variables?

File: src/goto-programs/goto_functions.h [hunk 2]
```diff
@@ -42,6 +43,12 @@ class goto_functionst
   typedef std::map<irep_idt, goto_functiont> function_mapt;
   function_mapt function_map;
 
+  // For coverage and multi-property
+  // Store and pass the coverage data in incr/kind mode
+  static std::unordered_set<std::string> reached_claims;
```
Comment (lucasccordeiro) [inline src/goto-programs/goto_functions.h:48]: Do we really need these static variables?

File: src/esbmc/bmc.cpp [hunk 3]
```diff
@@ -42,6 +40,10 @@
 #include <atomic>
 #include <goto-symex/witnesses.h>
 
+std::unordered_set<std::string> goto_functionst::reached_claims;
```
Comment (lucasccordeiro) [inline src/esbmc/bmc.cpp:43]: Do we really need these global variables?

**Review by lucasccordeiro — CHANGES_REQUESTED** [review]

File: src/goto-programs/goto_coverage.cpp (hunk 1 above)
Comment (ChenfengWei0) [reply src/goto-programs/goto_coverage.cpp:3]: It is the static member, must be defined and initialized once outside the class definition

File: src/goto-programs/goto_functions.h (hunk 2 above)
Comment (ChenfengWei0) [reply src/goto-programs/goto_functions.h:48]: It is the static member, must be defined and initialized once outside the class definition

File: src/esbmc/bmc.cpp (hunk 3 above)
Comment (ChenfengWei0) [reply src/esbmc/bmc.cpp:43]: It is the static member, must be defined and initialized once outside the class definition

**Review by rafaelsamenezes — APPROVED** [review]

**Review by lucasccordeiro — APPROVED** [review]

**Comment by lucasccordeiro** [comment]
Thanks for submitting this PR, @ChenfengWei0.

**Pull request merged**